cli/tests/pcluster/templates/test_cluster_stack.py

```diff
@@ -46,43 +46,127 @@ def test_cluster_builder_from_configuration_file(mocker, config_file_name):
+                        "HeadNodeRoleName": "",
+                        "ComputeFleetRoleNames": {"Ref": "Role15b342af42246b70"},
+                        "LaunchTemplate1f8c19f38f8d4f7fVersion": {
+                            "Fn::GetAtt": ["ComputeFleetLaunchTemplate1f8c19f38f8d4f7f3489FB83", "LatestVersionNumber"]
+                        },
+                        "LaunchTemplateA6f65dee6703df4aVersion": {
+                            "Fn::GetAtt": ["ComputeFleetLaunchTemplateA6f65dee6703df4a27E3DD2A", "LatestVersionNumber"]
+                        },
                     },
-                    "LaunchTemplate7916067054f91933Version": {
-                        "Fn::GetAtt": ["ComputeFleetLaunchTemplate7916067054f919332FB9590D", "LatestVersionNumber"]
+                },
+            },
+        ),
+        (
+            "byos-with-compute-fleet-instance-role.yaml",
+            {
+                "Type": "AWS::CloudFormation::Stack",
+                "Properties": {
+                    "TemplateURL": "https://parallelcluster-a69601b5ee1fc2f2-v1-do-not-delete.s3.fake-region.amazonaws."
+                    "com/parallelcluster/clusters/dummy-cluster-randomstring123/templates/"
+                    "byos-substack.cfn",
+                    "Parameters": {
+                        "ClusterName": "clustername",
+                        "ParallelClusterStackId": {"Ref": "AWS::StackId"},
+                        "VpcId": "vpc-123",
+                        "HeadNodeRoleName": "",
+                        "ComputeFleetRoleNames": "",
+                        "LaunchTemplate1f8c19f38f8d4f7fVersion": {
+                            "Fn::GetAtt": ["ComputeFleetLaunchTemplate1f8c19f38f8d4f7f3489FB83", "LatestVersionNumber"]
+                        },
+                        "LaunchTemplateA6f65dee6703df4aVersion": {
+                            "Fn::GetAtt": ["ComputeFleetLaunchTemplateA6f65dee6703df4a27E3DD2A", "LatestVersionNumber"]
+                        },
                     },
-                    "LaunchTemplateA46d18b906a50d3aVersion": {
-                        "Fn::GetAtt": ["ComputeFleetLaunchTemplateA46d18b906a50d3a347605B0", "LatestVersionNumber"]
+                },
+            },
+        ),
+        (
+            "byos.full.yaml",
+            {
+                "Type": "AWS::CloudFormation::Stack",
+                "Properties": {
+                    "TemplateURL": "https://parallelcluster-a69601b5ee1fc2f2-v1-do-not-delete.s3.fake-region.amazonaws."
+                    "com/parallelcluster/clusters/dummy-cluster-randomstring123/templates/"
+                    "byos-substack.cfn",
+                    "Parameters": {
+                        "ClusterName": "clustername",
+                        "ParallelClusterStackId": {"Ref": "AWS::StackId"},
+                        "VpcId": "vpc-123",
+                        "HeadNodeRoleName": "",
+                        "ComputeFleetRoleNames": {"Ref": "Role15b342af42246b70"},
```

Comment:
shall we test here multiple compute fleet role names?

Reply:
Do you mean multiple custom InstanceRole in compute queue? This compute fleet role is generated by pcluster and we only attach one role by default